apicapi/apic_client.py

```diff
@@ -403,63 +420,75 @@ def _send(self, request, url, data=None, refreshed=None, accepted=None,
                                          err_text=err_text, err_code=err_code)
         return imdata
 
+    def _get_request_call(self, request):
+        if request == ApicSession.HTTP_GET:
+            return self.session.get
+        elif request == ApicSession.HTTP_POST:
+            return self.session.post
+        elif request == ApicSession.HTTP_DELETE:
+            return self.session.delete
+        elif request == ApicSession.HTTP_PUT:
+            return self.session.put
+        else:
+            return None
```

Comment:
Why is the above method needed? It's a 1:1 mapping between session methods and class strings, that will require us to add a condition here for *every* new method we may want to use in the future! the "requests" library is likely not dropping support to these methods any time soon... I can see this being useful if we plan to replace the requests module, but it looks like more pain than anything for something that would bring many more issues if happens. Am I missing something?